### ISSUE TITLE
Report overwrite the same file

### ISSUE THREAD

**Issue by b4456609** [DiUS/pact-jvm#411]
Herer is my console output. I have two provider, theater and user. I also configure json and markdown report.

```
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:pactVerify_theater

Verifying a pact between order and theater
  [from URL http://192.168.5.101:8880/pacts/provider/theater/consumer/order/version/0.0.1]
  Given The show id'5ef2bf0d-6dbf-4de8-a095-93690854da5b' is exist and has 2 more seats
  Get la la land movie detail
    returns a response which
      has status code 200 (OK)
      includes headers
        "Content-Type" with value "application/json; charset=UTF-8" (OK)
      has a matching body (OK)
:pactVerify_user

Verifying a pact between order and user
  [from URL http://192.168.5.101:8880/pacts/provider/user/consumer/order/version/0.0.1]
  Given The user, Ben, is exists
  Get la la land movie detail
    returns a response which
      has status code 200 (OK)
      includes headers
        "Content-Type" with value "application/json; charset=UTF-8" (OK)
      has a matching body (OK)
:pactVerify

BUILD SUCCESSFUL

Total time: 3.532 secs
```
Only theater present in report directory.
```
.
├── build
│   └── reports
│       └── pact
│           ├── theater.json
│           └── theater.md
├── build.gradle
├── gradle
│   └── wrapper
│       ├── gradle-wrapper.jar
│       └── gradle-wrapper.properties
├── gradlew
├── gradlew.bat
├── Jenkinsfile
├── settings.gradle
└── uploadReport.sh

5 directories, 10 files
```
The content of the theater.md is the report of user provider.
I guess the report is overwrite the same file.
```
# user

| Description    | Value |
| -------------- | ----- |
| Date Generated | Sat Apr 22 10:41:00 CST 2017 |
| Pact Version   | 3.3.6 |

## Verifying a pact between _order_ and _user_

From URL: http://192.168.5.101:8880/pacts/provider/user/consumer/order/version/0.0.1
Given **The user, Ben, is exists**  
Get la la land movie detail  
&nbsp;&nbsp;returns a response which  
&nbsp;&nbsp;&nbsp;&nbsp;has status code **200** (<span style='color:green'>OK</span>)  
&nbsp;&nbsp;&nbsp;&nbsp;includes headers  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"**Content-Type**" with value "**application/json; charset=UTF-8**" (<span style='color:green'>OK</span>)  
&nbsp;&nbsp;&nbsp;&nbsp;has a matching body (<span style='color:green'>OK</span>)  
```
I use gradle wrapper version 2.13 and pact gradle plugin version 3.3.8.

**Comment by jannikolai** [comment]
I see the same behaviour in our project.
For the first executed test the files get created and are afterwards override by each test which is executed afterwards.
Changing it to just json or just markdown output did not have any consequence.

We are using `pact-jvm-provider-gradle_2.11:3.4.1`

**Comment by uglyog** [comment]
Fix has been released with version 3.5.18
